platform/xbox/functions/_PDCLIB/_PDCLIB_stdinit.c

```diff
@@ -358,10 +358,147 @@ struct _PDCLIB_lc_messages_t _PDCLIB_lc_messages = {
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
+        /* unknown error   */ (char *)"unknown error",
```

Comment:
nit: My man page says:

> The  strerror(),  strerror_l(),  and  the GNU-specific strerror_r() functions return the appropriate error description string, or an "Unknown error nnn" message if the error number is unknown.

I'm not sure if this is a Linux libc thing or C spec thing, but it's weird that the error number is not included here. Or are these values from the MS libc? (I did not look at the mentioned tool and also did not check what `strerror` does when the error number isn't part of the array).

Reply:
The C standard doesn't have that requirement, although I see the benefit. The "unknown error" message itself comes from the PDCLib tool itself.

Reply:
I feel like the tool should dump out `NULL` and then https://github.com/DevSolar/pdclib/blob/8e10fcf7fd3bac2c5640a86fa165e7fa8c411803/functions/string/strerror.c#L14 should be improved to handle those.
Then it would be obvious which messages are from the OS which provided the `strerror` messages, and which ones are from pdclib. It would also save on binary size (also interesting for embedded platforms).

*(Actually it's a bit weird, that the tool generates an array, instead of a switch/case, to avoid large gaps in error numbers)*

Wether the handler for unknown messages remains as is (constant string), or provides more context (could `sprintf()` a number and add new messages to a list at runtime) could be decided later then.

Reply:
I mentioned this discussion upstream https://github.com/DevSolar/pdclib/issues/9

I'd like to wait at least for a response, considering this is a low-priority PR (or is it required for something?).

Reply:
Hey there... I tinkered a bit with a different approach that removes the "holes" in the array. But that means each entry needs to hold the errno value it refers to, as I can no longer use the array index for the lookup.

That means each of the entries gets sizeof( int ) wider, for saving one sizeof( char * ) per "hole" value... already a marginal saving. And then I need a bit more code for the lookup... long story short, the patched build was **larger** than the current one by roundabout 300 bytes for a release build. I don't think that is what you're looking for...?!?

A patch tarball is available at http://rootdirectory.ddns.net/static/pdclib_patch_strerror.tgz if you want to experiment with that version.